### PR TITLE
DOC: Add macOS Homebrew installation instructions

### PR DESCRIPTION
**Pull request by fepegar**
Related to #5356.

@lassoan, I have added only the installation instructions, for now. Do you think the developer part is necessary? If yes, where should it go? I didn't find an obvious home for it in the Developer section.

### CODE REVIEW

**Comment by lassoan** [comment]
Thank you, this looks good to me. Information for developers should go somewhere here: https://slicer.readthedocs.io/en/latest/developer_guide/build_instructions/macos.html#package-slicer. Maybe you can add a section after "Package Slicer" section.

**Comment by fepegar** [comment]
Ok, thanks. Shall I add that to this PR?

**Comment by lassoan** [comment]
Yes, you can add that to this PR.

**Review by pieper — APPROVED** [review]
LGTM 👍

**Comment by pieper** [comment]
> The Lab on the wiki is now out of date. Shall I delete it?

I would say yes.  The less out of date info out there the less chance for confusion.

**Comment by pieper** [comment]
I believe the CI check is not related to this so we should override the check.  @lassoan any other comments on the PR?

**Comment by jamesobutler** [comment]
The CircleCI check has failed because the branch is not up-to-date with the Master branch and has differing files in the Superbuild directory. Rebasing would resolve the issue. 

The CommitCheck check is failing because `Add developer guide for Slicer Homebrew cask` doesn't have an appropriate commit prefix. It would need to be `DOC: Add developer guide for Slicer Homebrew cask` if it is going to be an individual commit and not squashed down.

**Comment by fepegar** [comment]
Thanks, James. Let me rebase and squash these.

Steve, I tried to remove the wiki page but couldn't find any buttons for that. Is it possible that I don't have permission to delete pages?

**Comment by pieper** [comment]
Yes probably you need permission for the wiki.  I can replace the page with a link to the new docs once they are ready.

for reference here's the page: https://www.slicer.org/wiki/Documentation/Labs/HomebrewCask

**Comment by fepegar** [comment]
Oh, I can do that, no problem. I have permission to edit. I just couldn't find a way to actually delete the page.

File: Docs/developer_guide/build_instructions/macos.md [hunk 1]
```diff
@@ -176,6 +176,28 @@ Library not loaded: /usr/local/opt/zstd/lib/libzstd.1.dylib
...
+
+Specifically, the `cask` extension is used, which allows management of graphical applications from the command line.
+Casks are Ruby files (`.rb`) describing the metadata necessary to download, check and install the package.
+
+The [cask for the Preview version of Slicer](https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/slicer-preview.rb) does not need maintenance, as it automatically downloads the latest binaries and therefore the exact version does not need to be specified in the cask.
+However, the [cask for the stable version](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/slicer.rb), which has traditionally been [maintained by Slicer users](https://github.com/Homebrew/homebrew-cask/commits/master/Casks/slicer.rb), needs to be manually updated every time a new stable version of Slicer is released.
+Instructions to update Homebrew casks can be found on the [`homebrew-cask` repository](https://github.com/Homebrew/homebrew-cask/commits/master/Casks/slicer.rb).
```
Comment (lassoan) [inline Docs/developer_guide/build_instructions/macos.md:188]: Is [this](https://github.com/Homebrew/homebrew-cask/blob/HEAD/CONTRIBUTING.md#updating-a-cask) the instruction you are referring to?

How do you determine the version number `4.11.20210226,1442768`?
Where does the `1442768` come from?

**Comment by lassoan** [comment]
Wiki pages cannot be deleted, only moved. We replace content of pages that are moved to readthedocs with a large banner, see for example here: https://www.slicer.org/wiki/Documentation/Nightly/Extensions/DebuggingTools?veaction=editsource

File: Docs/developer_guide/build_instructions/macos.md (hunk 1 above)
Comment (fepegar) [reply Docs/developer_guide/build_instructions/macos.md:188]: > Is this the instruction you are referring to?

Yes, thanks. That was a bad paste.

> Where does the 1442768 come from?

I suppose that the [bitstream ID](https://discourse.slicer.org/t/slicer-installation-on-mac-using-homebrew/933/19).

> How do you determine the version number 4.11.20210226,1442768?

I just look at the version on the download website and at its download link (https://download.slicer.org/bitstream/1442768). @jcfr provided in the link above a method to retrieve these automatically, if I understood correctly.

**Comment by lassoan** [comment]
Tnank you @fepegar for the documentation updates, the branch has been merged.

**Comment by fepegar** [comment]
Thanks for your guidance!

**Pull request merged**